orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -90,6 +90,24 @@ def _load_torch_allocator_cpp_extension(verbosity):
                        verbose=verbosity < Verbosity.WARNING, with_cuda=True)
 
 
+def _load_torch_rocm_allocator_cpp_extension(verbosity):
+    torch_rocm_allocator_addresses_cpp_source = """
+    #include <torch/extension.h>
+    #include <c10/hip/HIPCachingAllocator.h>
+    size_t rocm_caching_allocator_raw_alloc_address() {
+        return reinterpret_cast<size_t>(&c10::hip::HIPCachingAllocator::raw_alloc);
+    }
+    size_t rocm_caching_allocator_raw_delete_address() {
+        return reinterpret_cast<size_t>(&c10::hip::HIPCachingAllocator::raw_delete);
+    }
+    """
+
+    return load_inline(name='inline_extension', cpp_sources=[torch_rocm_allocator_addresses_cpp_source],
+                      extra_cflags=['-D__HIP_PLATFORM_HCC__=1'],
+                      functions=['rocm_caching_allocator_raw_alloc_address',
+                                  'rocm_caching_allocator_raw_delete_address'],
+                       verbose=verbosity < Verbosity.WARNING, with_cuda=True)
+
```

Comment:
Could you move this to a external_allocator.py along with the existing torch allocator? That would clean ortmodule.py a bit

Reply:
For frontend, I am thinking to have different code between ROCM and CUDA as less as possible. Can we rename cuda_caching_allocator_raw_alloc_address to gpu_caching_allocator_raw_alloc_address and share it between ROCm vs CUDA. We can just hide the difference in the string torch_cuda/rocm_allocator_addresses_cpp_source.